packages/headless/src/controllers/commerce/search-box/headless-search-box.test.ts

```diff
@@ -0,0 +1,272 @@
+      props.options!.highlightOptions = {
+        notMatchDelimiters: {
+          open: 1 as unknown as string,
+          close: 2 as unknown as string,
+        },
+      };
+      expect(() => initController()).toThrow();
+    });
+  });
+
+  it('should return the right state', () => {
+    expect(searchBox.state).toEqual({
+      value: state.querySet[id],
+      suggestions: state.querySuggest[id]!.completions.map((completion) => ({
+        highlightedValue: '<a>hi<a>light<i>ed<i>',
+        rawValue: completion.expression,
+      })),
+      isLoading: false,
+      isLoadingSuggestions: false,
+    });
+  });
+
+  it('should dispatch a registerQuerySetQuery action at initialization', () => {
+    expect(engine.actions).toContainEqual(
+      registerQuerySetQuery({id, query: state.commerceQuery.query!})
+    );
+  });
+
+  it('should dispatch a registerQuerySuggest action at initialization', () => {
+    expect(engine.actions).toContainEqual(
+      registerQuerySuggest({
+        id,
+      })
+    );
+  });
+
+  describe('when calling updateText', () => {
+    it('updates the search box query in the querySet', () => {
+      const text = 'query';
+      searchBox.updateText(text);
+
+      const action = updateQuerySetQuery({id, query: text});
+      expect(engine.actions).toContainEqual(action);
+    });
+
+    it('should call the showSuggestions method', () => {
+      jest.spyOn(searchBox, 'showSuggestions');
+      searchBox.updateText('how can i fix');
+
+      expect(searchBox.showSuggestions).toHaveBeenCalled();
+    });
+  });
+
+  it(`when calling clear
+      should dispatch a updateQuerySetQuery action`, () => {
+    searchBox.clear();
+    expect(engine.actions).toContainEqual(
+      updateQuerySetQuery({id: id, query: ''})
+    );
+  });
+
+  it(`when calling clear
+      should dispatch a clearQuerySuggest action`, () => {
+    searchBox.clear();
+    expect(engine.actions).toContainEqual(clearQuerySuggest({id}));
+  });
+
+  describe('#showSuggestions', () => {
+    it('dispatches fetchQuerySuggestions', async () => {
+      searchBox.showSuggestions();
+
+      const action = engine.actions.find(
+        (a) => a.type === fetchQuerySuggestions.pending.type
+      );
+      expect(action).toEqual(
+        fetchQuerySuggestions.pending(action!.meta.requestId, {id})
+      );
+    });
+  });
+
+  describe('#selectSuggestion', () => {
+    it('dispatches a selectQuerySuggestion action', () => {
+      const value = 'i like this expression';
+      searchBox.selectSuggestion(value);
+
+      expect(engine.actions).toContainEqual(
+        selectQuerySuggestion({id, expression: value})
+      );
+    });
+
+    it('dispatches executeSearch', () => {
+      const suggestion = 'a';
+      searchBox.selectSuggestion(suggestion);
+
+      expect(engine.findAsyncAction(executeSearch.pending)).toBeTruthy();
+    });
+  });
+
+  describe('when calling submit', () => {
+    it('it deselects all facets', () => {
```

Comment:
```suggestion
    it('dispatches #deselectAllBreadcrumbs', () => {
```

Reply:
Won't it only do that if `clearFilters` is `true`, though? We should test both cases.